### PR TITLE
Adding support for extensions

### PR DESCRIPTION
**Pull request by calvinatian**
### Summary

Adds ability query and change extension information for an assignment for a course

### Details

- [ ] Get extensions
- [x] Create new extensions
- [x] Modify existing extensions
- [ ] Remove extensions

### Checks
- [ ] Tested changes
- [ ] Attached Logs

### Team to Review

All

### Reference to the issue

None

### CODE REVIEW

**Comment by calvinatian** [comment]
Need to add an instructor test account to test changing extensions

**Comment by calvinatian** [comment]
> Need to add an instructor test account to test changing extensions

Updated secrets and tests with another account. There is now a student account, and an instructor account for the same test course: Environment variable names:

```
GRADESCOPE_CI_STUDENT_EMAIL
GRADESCOPE_CI_STUDENT_PASSWORD
GRADESCOPE_CI_INSTRUCTOR_EMAIL
GRADESCOPE_CI_INSTRUCTOR_PASSWORD
```

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -47,5 +47,7 @@ jobs:
       - name: Run Tests
         run: pdm test
         env:
-          EMAIL: ${{ secrets.GRADESCOPE_CI_STUDENT_EMAIL }}
-          PASSWORD: ${{ secrets.GRADESCOPE_CI_STUDENT_PASSWORD }}
+          GRADESCOPE_CI_STUDENT_EMAIL: ${{ secrets.GRADESCOPE_CI_STUDENT_EMAIL }}
```
Comment (jmargaret2) [inline .github/workflows/main.yaml:50]: I forget, where are we storing these secrets variables?

Comment (calvinatian) [reply .github/workflows/main.yaml:50]: They are stored in GitHub secrets in the repo settings. I'll also share the credentials on slack for local testing

Comment (aminoa) [reply .github/workflows/main.yaml:50]: Thanks Calvin! I'll wait for your dm

Comment (SushieeK) [reply .github/workflows/main.yaml:50]: Ctaching up with the missed Reviews.

**Review by SushieeK — COMMENTED** [review]
Catching up with the missed PR's

**Review by SushieeK — APPROVED** [review]
Catching up with the missed PR's

**Pull request merged**